### PR TITLE
Stacksets

### PR DESCRIPTION
**Pull request by tylersouthwick**
* **Please check if the PR fulfills these requirements**
- [x] The commit message describes your change
- [x] Tests for the changes have been added if possible (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] Changes are mentioned in the changelog (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

adds stacksets

* **What is the current behavior?** (You can also link to an open issue here)

no stackset support

* **What is the new behavior (if this is a feature change)?**
allows stackset manipulation

* **Does this PR introduce a breaking change?** (What changes might users need to make in their setup due to this PR?)
no

* **Other information**:

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -9,7 +9,7 @@
     </parent>
     <groupId>de.taimos</groupId>
     <artifactId>pipeline-aws</artifactId>
-    <version>1.23-SNAPSHOT</version>
+    <version>1.22-SNAPSHOT</version>
```
Comment (hoegertn) [inline pom.xml]: Why this?

File: README.md [hunk 2]
```diff
@@ -481,6 +481,9 @@ String result = invokeLambda(
 
 ## current master
 
+## 1.23
+* Add support for CloudFormation Stack Sets
```
Comment (hoegertn) [inline README.md]: Please leave under `current master` and I will change it when it is released

File: src/main/java/de/taimos/pipeline/aws/cloudformation/stacksets/AbstractCFNCreateStackSetStep.java [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+import hudson.model.TaskListener;
+import org.jenkinsci.plugins.workflow.steps.StepContext;
+import org.jenkinsci.plugins.workflow.steps.StepExecution;
+import org.kohsuke.stapler.DataBoundSetter;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.util.Collection;
+
+abstract class AbstractCFNCreateStackSetStep extends TemplateStepBase implements ParameterProvider {
```
Comment (hoegertn) [inline src/main/java/de/taimos/pipeline/aws/cloudformation/stacksets/AbstractCFNCreateStackSetStep.java]: TemplateStepBase already implements ParameterProvider

File: src/main/java/de/taimos/pipeline/aws/cloudformation/stacksets/CFNDeleteStackSetStep.java [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+import org.jenkinsci.plugins.workflow.steps.AbstractStepExecutionImpl;
+import org.jenkinsci.plugins.workflow.steps.AbstractStepImpl;
+import org.jenkinsci.plugins.workflow.steps.StepContextParameter;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+
+public class CFNDeleteStackSetStep extends AbstractStepImpl {
```
Comment (hoegertn) [inline src/main/java/de/taimos/pipeline/aws/cloudformation/stacksets/CFNDeleteStackSetStep.java]: Abstract* classes are deprecated

File: src/main/java/de/taimos/pipeline/aws/cloudformation/stacksets/CFNUpdateStackSetStep.java [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+
+public class CFNUpdateStackSetStep extends AbstractCFNCreateStackSetStep {
+
+	@DataBoundConstructor
+	public CFNUpdateStackSetStep(String stackSet) {
+		super(stackSet);
+	}
+
+	@Extension
+	public static class DescriptorImpl extends AbstractStepDescriptorImpl {
```
Comment (hoegertn) [inline src/main/java/de/taimos/pipeline/aws/cloudformation/stacksets/CFNUpdateStackSetStep.java]: deprecated

File: pom.xml (hunk 1 above)
Comment (tylersouthwick) [reply pom.xml]: typo from when I was testing it

**Comment by hoegertn** [comment]
- The changelog entry is now gone
- Please qualify with `this`
- Please avoid wildcard imports

I can fix the latter two after merging if you like

**Comment by tylersouthwick** [comment]
I merged #54 into this and it looks good. everything should be using 'this' qualified instance field/method access and no wildcards.

**Pull request merged**